### PR TITLE
MULE-9738 Improve functional test case to work with extensions - Migr…

### PR DESCRIPTION
**Pull request by gsfernandes**
…ate file extension to use isolated runner

### CODE REVIEW

File: extensions/file/pom.xml [hunk 1]
```diff
@@ -36,11 +36,6 @@
         </dependency>
 
         <dependency>
-            <groupId>com.google.guava</groupId>
-            <artifactId>guava</artifactId>
-        </dependency>
-
-        <dependency>
```
Comment (gsfernandes) [inline extensions/file/pom.xml]: It is already exposed by mule-core.

File: extensions/file/src/main/resources/META-INF/mule-module.properties [hunk 2]
```diff
@@ -1,4 +0,0 @@
-module.name=file
-
-artifact.export.classPackages=org.mule.extension.file.api
-artifact.export.resourcePackages=/META-INF
```
Comment (gsfernandes) [inline extensions/file/src/main/resources/META-INF/mule-module.properties]: No longer needed, any test that needs to use this extension should be migrated to use isolation runner. So far, not even integration tests are using this API for their tests.

File: extensions/file/src/test/java/org/mule/extension/file/FileConnectorTestCase.java [hunk 3]
```diff
@@ -62,12 +60,6 @@ protected void doTearDownAfterMuleContextDispose() throws Exception
         temporaryFolder.delete();
     }
 
-    @Override
-    protected Class<?>[] getAnnotatedExtensionClasses()
-    {
-        return new Class<?>[] {FileConnector.class};
-    }
-
```
Comment (gsfernandes) [inline extensions/file/src/test/java/org/mule/extension/file/FileConnectorTestCase.java]: Now extensions are auto-discovered by the runner and their metadata is generated while classloader is built.

**Comment by gsfernandes** [comment]
Ready to be reviewed!

File: extensions/file/src/test/resources/log4j2-test.xml [hunk 4]
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+  ~ The software in this package is published under the terms of the CPAL v1.0
```
Comment (pablokraan) [inline extensions/file/src/test/resources/log4j2-test.xml]: We don't add license header to XML files

**Pull request merged**